packages/xod-project/src/flatten.js

```diff
@@ -437,14 +437,26 @@ const getValueToBind = R.curry((nodesById, linksChain) => {
     //    +---O-----------+
     //    |  ...          |
     //    +---------------+
-    : findBoundValue(R.find, R.init(outputNodesFromLinkChain)).map((bottomBoundValue) => {
-      const topLink = R.last(linksChain);
-      return {
-        nodeId: Link.getLinkOutputNodeId(topLink),
-        pinKey: Link.getLinkOutputPinKey(topLink),
-        value: bottomBoundValue,
-      };
-    });
+    : findBoundValue(R.find, R.init(outputNodesFromLinkChain))
+      .chain((bottomBoundValue) => {
+        const pinKey = Link.getLinkOutputPinKey(R.last(linksChain));
+        const topNode = R.last(outputNodesFromLinkChain);
+        const maybeBoundValue = Node.getBoundValue(pinKey, topNode);
+
+        // if some value is already bound to the
+        // top node's output — don't overwrite it
+        return maybeBoundValue.isJust
+          ? Maybe.Nothing()
+          : Maybe.Just(bottomBoundValue);
```

Comment:
Use `foldMaybe`? It would survive possible (?) `Maybe` lib replacement.

Reply:
Done 👍🏻